dask/dataframe/core.py

```diff
@@ -1523,6 +1523,14 @@ def query(self, expr, **kwargs):
                                  dummy, self.divisions)
 
     @derived_from(pd.DataFrame)
+    def eval(self, expr, inplace=None, **kwargs):
+        if '=' in expr and inplace:
```

Comment:
you need to explicitly verify that `inplace is False`, because with the current default behavior `inplace=None` implies `inplace=True`.